Update: More Bank Location changes

gmason0 opened the pull request:
I have added some more checks for the hasRequirements to some of the skilling guild banks that are listed in the bank locations to ensure that players aren't sent somewhere they can't access.

I have also added the Rouge's Den Chest along side of renaming the existing Rouge's Den Bank to Emerald Benedict.

With this, I have also created two new methods inside of Rs2Player called getRealSkillLevel & getBoostedSkillLevel, not really needed but does save calling an additional getter when calling for these methods.

The only guild that has a bank that I am missing is Legends Guild, if we could add that to this as well, that would be great. The check would be as follows.

```
case LEGENDS_GUILD:
    if (hasLineOfSight) return true;
    return Rs2Player.getQuestState(Quest.LEGENDS_QUEST) == QuestState.FINISHED;
```

gmason0 committed:
Update: More Bank Location changes (#8)

* feat: add getRealSkillLevel & getBoostedSkillLevel

* update: added more hasRequirement checks for skilling guilds

gmason0 committed:
hotfix: missed woodcutting guild

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/enums/BankLocation.java:
```diff
@@ -99,19 +101,38 @@ public boolean hasRequirements() {
                 return Rs2Player.isMember() && isWearingCooksGuild && (hasVarrockHardDiary || hasMaxedCooking);
+            case WARRIORS_GUILD:
+                if (hasLineOfSight) return true;
+                return (Rs2Player.getRealSkillLevel(Skill.ATTACK) >= 99 || Rs2Player.getRealSkillLevel(Skill.STRENGTH) >= 99) ||
```
I think warriors guild is missing a Rs2Player.isMember check

gmason0 replied:
thats right, it is!

https://oldschool.runescape.wiki/w/Guilds

gmason0 replied:
The checks for Warrior's Guild are now there in the latest commit c54c442

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/enums/BankLocation.java:
It might be a good idea to create a method checkRequirement(Skill skill, int levelRequired) and wrap common code in there. This will avoid duplicate code and makes it easy to add more to it if needed

gmason0 replied:
Will do!

gmason0 replied:
In commit c54c442, i created a third method called checkSkillRequirement, I still left getRealSkillLevel & getBoostedSkillLevel to be used to pull the skill level that is used for line 121 in BankLocation.java 

```
 (Rs2Player.getRealSkillLevel(Skill.ATTACK) + Rs2Player.getRealSkillLevel(Skill.STRENGTH) >= 130);
 ```

gmason0 replied:
This has been updated to include legend's guild & I have also added a couple more Rs2Player Methods to get Ironman Status.

gmason0 replied:
I have changed the requirements for the crafting guild, the requirements to enter the crafting guild are only 40 crafting & the apron but to use the bank in the crafting guild has additional requirements.

https://oldschool.runescape.wiki/w/Crafting_Guild#Bank_chest

gmason0 committed:
chore: changes per PR comments

gmason0 committed:
refactor: Rs2Player.checkSkillRequirement() -> Rs2Player.getSkillRequirement()
feat: add Rs2Player.isIronman() & RS2Player.isGroupIronman()
update: add Legend's Guild Bank & requirements for that bank

gmason0 committed:
hotfix: adjust getSkillRequirement to use Rs2Player methods

gmason0 committed:
update: add clay rocks to Auto Miner

gmason0 committed:
fix: adjust requirements for crafting guild bank

gmason0 committed:
hotfix: further adjustments to the logic check for crafting guild & cooks guild
